x/epochs/abci.go

```diff
@@ -29,11 +29,13 @@ func BeginBlocker(ctx sdk.Context, k keeper.Keeper) {
 			epochStart = true
 			epochInfo.CurrentEpoch = epochInfo.CurrentEpoch + 1
 		}
-
 		if epochStart {
 			epochInfo.CurrentEpochEnded = false
-			epochInfo.CurrentEpochStartTime = ctx.BlockTime()
-
+			if epochInfo.CurrentEpochStartTime.Equal(time.Time{}) { //genesis state, initialize
+				epochInfo.CurrentEpochStartTime = epochInfo.StartTime
+			} else {
+				epochInfo.CurrentEpochStartTime = epochInfo.CurrentEpochStartTime.Add(epochInfo.Duration)
```

Comment:
* Main Logic to fix epoch time drift

Currently, the time drift is taking place because epoch start time is set to the very time when the logic is being executed. 
To fix this problem, epoch start time should be resulted by adding duration to the previous epoch start time

Reply:
Ok awesome. 

An idea that just occured to me is that maybe we should log that were starting a new epoch as well. (Using ctx.Logger().Info if I remember correctly)

Reply:
Bump on adding a ctx.logger here

Reply:
It only seems to be done in the uninitialized case atm?